### PR TITLE
Fixes #9919: Cleaned up postgres lineage and fixed function in postgres query_parser

### LINKED ISSUE #9919: Cleanup postgres lineage and usage
The postgres/lineage.py file is not following the same approach as the rest of lineage sources ([example](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/database/postgres/lineage.py#L49)).

The [process_table_query](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/database/postgres/query_parser.py#L137) is not returning the same object that it says it does.

We should try to align as much as possible this logic here taking Snowflake as an example

### PR DESCRIPTION
**Pull request by TomBushell**
### Describe your changes:

Targets #9919.

Cleaned up postgres/lineage.py and updated postgres/process_table_query.py to return None when there are no queries.

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
@TomBushell Thanks for the contribution
@pmbrull @ulixius9 can we merge this

File: ingestion/src/metadata/ingestion/source/database/postgres/lineage.py [hunk 1]
```diff
@@ -55,19 +57,8 @@ def _iter(self, *_, **__) -> Iterable[Either[AddLineageRequest]]:
...
-                    schema_name=table_query.databaseSchema,
-                    dialect=Dialect.POSTGRES,
-                )
 
-                for lineage_request in lineages or []:
-                    yield lineage_request
+        yield from super()._iter()
 
         if self.service_connection.scheme == PostgresScheme.pgspider_psycopg2:
             lineages = get_lineage_from_multi_tenant_table(
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/database/postgres/lineage.py:64]: changed the code as we would still need the processing for pgspider

**Comment by pmbrull** [comment]
Hi @TomBushell there's a python test that needs to be fixed before we can merge this in. Let us know if you need any help.

Thanks

**Comment by pmbrull** [comment]
@ulixius9 if you have a few minutes let's go over the test and see if we still want to bring this in

**Review by ayush-shah — APPROVED** [review]

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**